oss_src/unity/python/sframe/data_structures/sarray.py

```diff
@@ -1952,12 +1952,42 @@ def sample(self, fraction, seed=None):
         if (self.size() == 0):
             return SArray()
         if seed is None:
-            seed = time.time()
+            seed = abs(hash("%0.20f" % time.time())) % (2 ** 31)
 
 
         with cython_context():
             return SArray(_proxy=self.__proxy__.sample(fraction, seed))
 
+    def hash(self, seed=0):
+        """
+        Returns an SArray with a hash of each element. seed can be used
+        to change the hash function to allow this method to be used for
+        random number generation.
+
+        Parameters
+        ----------
+        seed : int
+            Defaults to 0. Can be changed to different values to get 
+            different hash results.
+
+        Returns
+        -------
+        out : SArray
+            An integer SArray with a hash value for each element. Identical
+            elements are hashed to the same value
+        """
+        with cython_context():
+            return SArray(_proxy=self.__proxy__.hash(seed))
+
+    @classmethod
+    def random_integers(self, size, seed=None):
```

Comment:
When it's a classmethod, the first parameter shouldn't be `self`, usually `cls` is used.